### DISCUSSION TITLE
Unable to retrieve changelog for Passbolt (docker datasource) who uses a custom versioning

### DISCUSSION THREAD

**Discussion by MindTooth** [renovatebot/renovate#24564] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Being able to retrieve changelog using `customChangelogUrl` because the container images are missing **org.opencontainers.image.source**.

Image: https://hub.docker.com/r/passbolt/passbolt
Repo: https://github.com/passbolt/passbolt_api

### What you tried so far.

Adding a custom changelog declaration:

```json
  "packageRules": [
    {
      "matchPackageNames": ["passbolt/passbolt"],
      "customChangelogUrl": "https://github.com/passbolt/passbolt_api"
    }
  ]
```

Repro: https://github.com/MindTooth/renovate-passbolt-test/pull/2

Run: https://developer.mend.io/github/MindTooth/renovate-passbolt-test/-/job/9a287170-0ab7-42d2-a94e-c7960127faa1

It detetects the custom URL, but says **No releases**.  I don't understand why. 🤷🏻‍♂️ 

```
DEBUG: Fetching changelog: https://github.com/passbolt/passbolt_api (4.0.0 -> 4.2.0) (branch="renovate/passbolt-passbolt-4.x")
DEBUG: No releases (branch="renovate/passbolt-passbolt-4.x")
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Looks like it's because they use silly names for releases. I _thought_ Renovate might still look at the tags. You're welcome to try to debug it and find out more, but otherwise supporting such an edge case isn't a high priority for us.

**Reply by MindTooth** [reply]
I can try more. :)  Been at it for hours already trying to read the source code and various attempts at `extractVersion`/`versioning`, etc.  I thought at least `4.0.0 -> 4.2.0` would mean looking at tags/releases between 4.0.0 and 4.2.0. 🤷🏻‍♂️  So for me, it seems that the output is not what it looks up?

Thanks for a quick answer. 👍🏻

**Reply by rarkins** [reply]
I thought that Renovate looks at the release name but also falls back to the tag name. In this case it doesn't seem to be the case, but I'm not sure why

**Reply by viceice** [reply]
renovate uses the current datasource to query versions before it uses them to fetch changelogs.

**Reply by MindTooth** [reply]
Two things:

1. Would the debug message then be wrong?  As it's comparing something different (I guess `4.1.0-1-ce` vs `4.2.0` (`newValue`)).
2. Could a feature like `changelogVersioning` be useful?

Thanks to both.

**Reply by viceice** [reply]
There is already a feature request to support a extract version for change logs

**Reply by MindTooth** [reply, marked as answer]
I guess you refer to https://github.com/renovatebot/renovate/issues/7400.  I'll resolve this on to your issue.  Thanks.
